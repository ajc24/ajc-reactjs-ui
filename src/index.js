export { default as Footer } from './ui-elements/Footer';
export { default as Header } from './ui-elements/Header';
export { default as HeaderLogo } from './ui-elements/HeaderLogo';
export { default as HeaderTitleAndSubtitle } from './ui-elements/HeaderTitleAndSubtitle';
export { default as MainContentPageTitle } from './ui-elements/MainContentPageTitle';
export { default as MainContentPanel } from './ui-elements/MainContentPanel';
export { default as MainContentText } from './ui-elements/MainContentText';
export { default as Menu } from './ui-elements/Menu';
export { default as Screenshot } from './ui-elements/Screenshot';
export { default as SmallMenuIcon } from './ui-elements/SmallMenuIcon';
export { default as SmallRightArrowIcon } from './ui-elements/SmallRightArrowIcon';
