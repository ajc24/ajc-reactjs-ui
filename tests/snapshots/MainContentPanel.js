import React from 'react';
import { TestDev } from 'ajc-jest-enzyme';
import { MainContentPanel } from '../../src';

describe('MainContentPanel', () => {
  describe('Default props and rendering - Component with white background', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <MainContentPanel title="Test Title">
            This is a test.
          </MainContentPanel>
        </React.Fragment>
      );
    });
    
    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Component with grey background', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <MainContentPanel title="Test Title" colour="grey">
            This is a test.
          </MainContentPanel>
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Component with transparent background', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <MainContentPanel title="Test Title" colour="transparent">
            This is a test.
          </MainContentPanel>
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Component with yellow background', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <MainContentPanel title="Test Title" colour="yellow">
            This is a test.
          </MainContentPanel>
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });
});
