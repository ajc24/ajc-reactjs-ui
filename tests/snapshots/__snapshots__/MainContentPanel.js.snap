// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContentPanel Default props and rendering - Component with white background verifies the snapshot for the component 1`] = `
<div
  className="ajc-main-content-container ajc-text-black ajc-default ajc-background-white"
  role="region"
>
  <div
    className="ajc-main-content-outer"
  >
    <div
      className="ajc-main-content-inner"
    >
      This is a test.
    </div>
  </div>
</div>
`;

exports[`MainContentPanel Transferred props and rendering - Component with grey background verifies the snapshot for the component 1`] = `
<div
  className="ajc-main-content-container ajc-text-black ajc-default ajc-background-grey-1"
  role="region"
>
  <div
    className="ajc-main-content-outer"
  >
    <div
      className="ajc-main-content-inner"
    >
      This is a test.
    </div>
  </div>
</div>
`;

exports[`MainContentPanel Transferred props and rendering - Component with transparent background verifies the snapshot for the component 1`] = `
<div
  className="ajc-main-content-container ajc-text-black ajc-default ajc-background-transparent"
  role="region"
>
  <div
    className="ajc-main-content-outer"
  >
    <div
      className="ajc-main-content-inner"
    >
      This is a test.
    </div>
  </div>
</div>
`;

exports[`MainContentPanel Transferred props and rendering - Component with yellow background verifies the snapshot for the component 1`] = `
<div
  className="ajc-main-content-container ajc-text-black ajc-default ajc-background-yellow"
  role="region"
>
  <div
    className="ajc-main-content-outer"
  >
    <div
      className="ajc-main-content-inner"
    >
      This is a test.
    </div>
  </div>
</div>
`;
